AsyncDisplayKit/ASDisplayNodeExtras.mm

```diff
@@ -10,6 +10,18 @@
 #import "ASDisplayNodeInternal.h"
 #import "ASDisplayNode+FrameworkPrivate.h"
 
+extern ASInterfaceState ASInterfaceStateForDisplayNode(ASDisplayNode *displayNode, UIWindow *window)
+{
+    if (displayNode && [displayNode supportsRangeManagedInterfaceState]) {
+        // Only use the interfaceState of nodes that are range managed
+        ASInterfaceState interfaceState = displayNode.interfaceState;
+        return (window == nil ? (interfaceState &= (~ASInterfaceStateVisible)) : interfaceState);
```

Comment:
Now that this has been generalized to work for all ASDisplayNodes, we need to assert !layerBacked, as a node that is layerBacked may have a nil window.  Fortunately, the current use cases will never hit this as they must always be view backed, which means that all of their containers must also be view backed.